tb_pulumi/ci.py

```diff
@@ -113,256 +109,199 @@ def __init__(
+                            'ecr:InitiateLayerUpload',
+                            'ecr:GetDownloadUrlForLayer',
+                            'ecr:ListImages',
+                            'ecr:UploadLayerPart',
+                            'ecr:PutImage',
+                        ],
+                        'Resource': [
+                            f'arn:aws:ecr:{project.aws_region}:{project.aws_account_id}:repository/{repo}*'
+                            for repo in ecr_repositories
+                        ],
+                    },
+                    {
+                        'Sid': 'AuthActions',
+                        'Effect': 'Allow',
+                        'Action': ['ecr:GetAuthorizationToken'],
+                        'Resource': ['*'],
+                    },
+                ],
+            }
+            policy_json = json.dumps(policy_dict)
+            ecr_image_push_policy = aws.iam.Policy(
+                f'{name}-policy-ecrpush',
+                name=f'{name}-ecr-push',
+                description=f'Allows CI automation for {project.project} to push container images to ECR.',
+                policy=policy_json,
                 opts=pulumi.ResourceOptions(parent=self),
+                tags=self.tags,
             )
 
-            access_key = aws.iam.AccessKey(
-                f'{name}-accesskey',
-                user=user.name,
-                opts=pulumi.ResourceOptions(parent=self, depends_on=[user]),
+        if enable_s3_bucket_upload:
+            policy_resources = []
+            for bucket in s3_upload_buckets:
+                policy_resources.extend([f'arn:aws:s3:::{bucket}', f'arn:aws:s3:::{bucket}/*'])
+            policy_dict = {
+                'Version': '2012-10-17',
+                'Statement': [
+                    {
+                        'Sid': 'S3BucketAndObjectAccess',
+                        'Effect': 'Allow',
+                        'Action': ['s3:PutObject', 's3:ListBucket'],
+                        'Resource': policy_resources,
+                    }
+                ],
+            }
+            policy_json = json.dumps(policy_dict)
+            s3_upload_policy = aws.iam.Policy(
+                f'{name}-policy-s3upload',
+                name=f'{name}-s3-upload',
+                description=f'Allows CI automation for {project.project} to upload files to certain S3 buckets.',
+                policy=policy_json,
+                opts=pulumi.ResourceOptions(parent=self),
+                tags=self.tags,
             )
-
-            secret_value = pulumi.Output.all(id=access_key.id, secret=access_key.secret).apply(
-                lambda args: json.dumps({'aws_access_key_id': args['id'], 'aws_secret_access_key': args['secret']})
+        if enable_full_s3_access:
+            resources = []
+            for bucket in s3_full_access_buckets:
+                resources.append(f'arn:aws:s3:::{bucket}')
+                resources.append(f'arn:aws:s3:::{bucket}/*')
```

Comment:
I guess this is to prevent accidental access to like `MyCoolBucket123` when you're trying to give it full access to just `MyCoolBucket`?